comp/core/workloadmeta/collectors/internal/containerd/image_sbom_trivy.go

```diff
@@ -31,21 +41,31 @@ func (c *collector) startSBOMCollection(ctx context.Context) error {
 
 	c.scanOptions = sbom.ScanOptionsFromConfig(config.Datadog, true)
 	c.sbomScanner = scanner.GetGlobalScanner()
+	c.queue = workqueue.NewRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(baseBackoffTime, maxBackoffTime))
```

Comment:
I feel like the `queue` should be part of the `scanner.Scanner` for simplicity. 
* We request container_image scan and the scanner.Scanner get the request, and if it succeed return a result in the results channel. (for each SBOM scanner error, it return an error in the channel).


basically we can replace the `scanner.scanQueue` by the new queue. https://github.com/DataDog/datadog-agent/blob/ea7d79be5f4438de2ca505cbc067b8c79be26480/pkg/sbom/scanner/scanner.go#L54C2-L54C11

Reply:
I think it could work but it would mean that:
- We need to tell the scanner about unset events so it stops retrying.
- Avoid using the rate limiter for host scans.